File: Baekjoon/Problem_1000~1999/Problem_1475/main.py
import sys

input = lambda: sys.stdin.readline().rstrip()
    
if __name__ == "__main__":
    N = [ int(n) for n in input()]

    counts = [ 0 ] * 10
    for n in N:
        cur = n
        
        # 6과 9 balancing
        if (cur == 6 or cur == 9) :
            if (counts[6] > counts[9]): cur = 9
            else: cur = 6
        
        counts[cur] += 1
        
    print(max(counts))

File: Baekjoon/Problem_1000~1999/Problem_1475/main.cpp
#include <cstdio>
#include <string>
#include <algorithm>

const int MAXN = 7;
const int MAXCNT = 10;

using namespace std;

int main() {
    char N[MAXN+1]; scanf("%s", N);
    string strN = string(N);

    int counts[MAXCNT] = { 0, };

    for (char n: strN) {
        int cur = n - '0';

        // 6 and 9 balancing
        if (cur == 6 || cur == 9) {
            if (counts[6] > counts[9]) cur = 9;
            else cur = 6;
        }

        counts[cur]++;
    }

    printf("%d\n", *max_element(counts, counts+MAXCNT));

    return 0;
}